File: Round-1/Array/Shuffle the array/shuffle_the_array.cpp
// Sol - 1 > My approach to suffle the array time complexity is O(n)
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> shuffle(vector<int> &nums, int n)
    {
        vector<int> result;
        for (uint32_t i = 0, c = n; i < n; ++i, ++c)
        {
            result.push_back(nums[i]);
            result.push_back(nums[c]);
        }
        return result;
    }
};
int main()
{
    Solution shu;
    vector<int> nums{1, 2, 3, 4, 5, 6, 7, 8, 0, 9};
    int len = nums.size() / 2;
    shu.shuffle(nums, len);
    return 0;
}

File: Round-1/Array/Shuffle the array/shuffle_the_array.js
// Sol - 1 > My approach to suffle the array time complexity is O(n)
var shuffle = function (nums, n) {
    let center = nums.length / 2;
    let initial = 0;
    const result = [];
    for (let i = initial, c = center; i < center; i++, c++) {
        result.push(nums[i], nums[c]);
    }
    return result;
};

const arr = [1, 2, 4, 5, 6, 7, 4, 3, 3, 6, 4];
console.log(shuffle(arr, arr.length / 2));